### DISCUSSION TITLE
Deprecation of executor options to generate package.json for libraries

### DISCUSSION THREAD

**Discussion by MichaelBorde** [nrwl/nx#18445] [General]
I would like to understand why some options to generate a package.json are now deprecated.

The related PR is here: https://github.com/nrwl/nx/pull/17923

In [documentation](https://nx.dev/packages/js/executors/tsc#buildableprojectdepsinpackagejsontype) I can see:

> Configure the project to use the '@nx/dependency-checks' ESLint rule instead (https://nx.dev/packages/eslint-plugin/documents/dependency-checks). It will be removed in v17.

I don't understand how a linter could create a package.json.
You might mean that I can manually create a project own package.json and verify the libraries versions are correct based on monorepo main package.json.

What are your motivations?

**Comment by Chanki-Min** [comment]
Same thought. Is Nx's [single version policy](https://nx.dev/concepts/integrated-vs-package-based#integrated-repos) is cope with dependency-checks?
